internal/pkg/template/template.go

```diff
@@ -92,10 +130,48 @@ func (c *Content) MarshalBinary() ([]byte, error) {
 	return c.Bytes(), nil
 }
 
+// newTemplate returns a named text template with the "indent" and "include" functions.
+func newTemplate(name string) *template.Template {
+	t := template.New(name)
+	t.Funcs(map[string]interface{}{
+		"include": func(name string, data interface{}) (string, error) {
+			// Taken from https://github.com/helm/helm/blob/8648ccf5d35d682dcd5f7a9c2082f0aaf071e817/pkg/engine/engine.go#L147-L154
+			buf := bytes.NewBuffer(nil)
+			if err := t.ExecuteTemplate(buf, name, data); err != nil {
+				return "", err
+			}
+			return buf.String(), nil
+		},
+		"indent": func(spaces int, s string) string {
+			// Taken from https://github.com/Masterminds/sprig/blob/48e6b77026913419ba1a4694dde186dc9c4ad74d/strings.go#L109-L112
+			pad := strings.Repeat(" ", spaces)
+			return pad + strings.Replace(s, "\n", "\n"+pad, -1)
+		},
+	})
+	return t
+}
+
 func (t *Template) read(path string) (string, error) {
 	s, err := t.box.FindString(path)
 	if err != nil {
 		return "", fmt.Errorf("read template %s: %w", path, err)
 	}
 	return s, nil
 }
+
+func (t *Template) parseTemplate(name, path string, options ...ParseOption) (*template.Template, error) {
```

Comment:
Does this method need to be attached to the Template struct?  I see that it needs the reference to `t` in the call to `read()` but is there any drawback to generating a new Template within a method attached to a Template? 

Or is this just semantic confusion on my part between the "template parser" and the "parsed template?"

Reply:
ah yeah this is definitely confusing lol. So `t` refers to our own `Template` object but `newTemplate` returns a `text/template` object 😛 so there are two different `Template` types in this pkg.

I'll rename the methods this way, do you think it makes it clearer?  

`parseTemplate -> parse` and add the comment `// parse reads the file at path and return a text/template object with the given name`

 `newTemplate -> newTextTemplate`

Reply:
Yes, that clarifies things nicely. I think in this ambiguous case it's good to disambiguate by using the short name for the object defined in the current file or package and the longer name for external objects with the same name.